recipes/azure-core-cpp/meta.yaml

```diff
@@ -0,0 +1,58 @@
+{% set name = "azure-core-cpp" %}
+{% set version = "1.10.0" %}
+{% set sha256 = "e6ae26ad30376bef6272024564764ba7bf5756063dc94cc4b74180cd004ab75e" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/Azure/azure-sdk-for-cpp/archive/refs/tags/azure-core_{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler("cxx") }}
+    - cmake
+    - make  # [not win]
+    - ninja  # [win]
+  host:
+    - libcurl
+    - openssl  # [not win]
+    - wil  # [win]
+  run:
+    - libcurl
+    - openssl  # [not win]
+    - wil  # [win]
```

Comment:
This whole section shouldn't be needed. The first two come via run_exports and the last one is a header only library.

Reply:
I don't think 2 is an option because it would make `azure-core-cpp` unusable by itself.

Reply:
After discussing with an upstream Azure SDK developer, we decided to apply a patch to remove the exported dependency on `wil` in the config file for `azure-core-cpp`

For more context, see https://github.com/Azure/azure-sdk-for-cpp/issues/4784 and https://github.com/Azure/azure-sdk-for-cpp/pull/4785